orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -1359,3 +1372,49 @@ def forward(self, mask, x1, x2):
 
     assert y1 is not None
     assert y2 is not None and y2.dtype == torch.uint8
+
+def test_model_partially_requires_grad():
+    device = 'cuda'
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetPartialNoGradModel(D_in, H, D_out).to(device)
+    model = ORTModule(model)
+    x = torch.randn(N, D_in, device=device)
+
+    # Make sure no exception is raised
+    output = model(x)
+
+    loss = torch.sum(output)
+    loss.backward()
+
+def test_model_wrapped_inside_torch_no_grad():
+    device = 'cuda'
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetSinglePositionalArgument(D_in, H, D_out).to(device)
+    model = ORTModule(model)
+    x = torch.randn(N, D_in, device=device)
+
+    # Make sure no exception is raised
+    with torch.no_grad():
+        output = model(x)
+
+def test_model_initializer_requires_grad_changes_from_one_forward_to_next():
+    device = 'cuda'
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetPartialNoGradModel(D_in, H, D_out).to(device)
+    model.fc1.requires_grad_(True)
+    model = ORTModule(model)
+    x = torch.randn(N, D_in, device=device)
+
+    # Make sure no exception is raised
+    output = model(x)
+    loss = torch.sum(output)
+    loss.backward()
+    training_session1 = model._training_session
+
+    model._original_module.fc1.requires_grad_(False)
+    output = model(x)
+    loss = torch.sum(output)
+    loss.backward()
```

Comment:
Is there some way we could test that this is having the intended effect? e.g. Before the first call, assert that `model.fc1.grad` is zero, then non-zero after the first call, and the same non-zero value after the second call? (I'm assuming that `requires_grad_(False)` doesn't release the accumulated grad buffer....)

Reply:
I added the asserts to test that.